### PR TITLE
oneMKL samples for beta09

### PR DESCRIPTION
**Pull request by petercad**
# Description

This PR contains six new oneMKL samples for beta09, from the oneMKL development team.

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [X] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [ ] Implement coding guidelines and ensure code quality. (*more polishing can be done*)
- [X] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [X] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [X] Adhere to readme template 
- [X] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [ ] Enforce format via clang-format config file
- [x] Review DPC++ code with Paul Peterson. (GitHub User: pmpeter1)
- [x] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [ ] Tested using Dev Cloud when applicable
- [ ] Implement fixes for ONSAM Jiras (*n/a*)
- [ ] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

**Comment by petercad** [comment]
@pmpeter1 would you be able to review the DPC++ code? There are still parts of the code that can be rewritten to be more idiomatic for C++, but I thought it was better to start the PR process now.

@JoeOster would you be able take a look at the READMEs?

File: Libraries/oneMKL/block_cholesky_decomposition/sample.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "Block Cholesky Decomposition",
```
Comment (JoeOster) [inline Libraries/oneMKL/block_cholesky_decomposition/sample.json:2]: **Add a sample GUID** Go here : https://www.guidgenerator.com/ Checkbox 'Uppercase' and 'hyphens' field Copy the value that is like : 825BEB2D-C9FA-4527-8770-9F7D765B65AB

In your sample.json, enter a field called "guid" at the very top and make it :

> "guid" : "825BEB2D-C9FA-4527-8770-9F7D765B65AB"

File: Libraries/oneMKL/block_lu_decomposition/sample.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "Block LU Decomposition",
```
Comment (JoeOster) [inline Libraries/oneMKL/block_lu_decomposition/sample.json:2]: **Add a sample GUID** Go here : https://www.guidgenerator.com/ Checkbox 'Uppercase' and 'hyphens' field Copy the value that is like : 825BEB2D-C9FA-4527-8770-9F7D765B65AB

In your sample.json, enter a field called "guid" at the very top and make it :

> "guid" : "825BEB2D-C9FA-4527-8770-9F7D765B65AB"

File: Libraries/oneMKL/monte_carlo_pi/sample.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "Monte Carlo Pi Estimation",
```
Comment (JoeOster) [inline Libraries/oneMKL/monte_carlo_pi/sample.json:2]: **Add a sample GUID** Go here : https://www.guidgenerator.com/ Checkbox 'Uppercase' and 'hyphens' field Copy the value that is like : 825BEB2D-C9FA-4527-8770-9F7D765B65AB

In your sample.json, enter a field called "guid" at the very top and make it :

> "guid" : "825BEB2D-C9FA-4527-8770-9F7D765B65AB"

File: Libraries/oneMKL/sparse_conjugate_gradient/sample.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "Sparse Conjugate Gradient",
```
Comment (JoeOster) [inline Libraries/oneMKL/sparse_conjugate_gradient/sample.json:2]: **Add a sample GUID** Go here : https://www.guidgenerator.com/ Checkbox 'Uppercase' and 'hyphens' field Copy the value that is like : 825BEB2D-C9FA-4527-8770-9F7D765B65AB

In your sample.json, enter a field called "guid" at the very top and make it :

> "guid" : "825BEB2D-C9FA-4527-8770-9F7D765B65AB"

File: Libraries/oneMKL/black_scholes/sample.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "Black-Scholes",
```
Comment (JoeOster) [inline Libraries/oneMKL/black_scholes/sample.json:2]: **Add a sample GUID** Go here : https://www.guidgenerator.com/ Checkbox 'Uppercase' and 'hyphens' field Copy the value that is like : 825BEB2D-C9FA-4527-8770-9F7D765B65AB

In your sample.json, enter a field called "guid" at the very top and make it :

> "guid" : "825BEB2D-C9FA-4527-8770-9F7D765B65AB"

File: Libraries/oneMKL/block_cholesky_decomposition/sample.json (hunk 1 above)
Comment (petercad) [reply Libraries/oneMKL/block_cholesky_decomposition/sample.json:3]: Done, thanks!

**Comment by JoeOster** [comment]
@petercad - please take a look at the ci errors

**Comment by petercad** [comment]
@JoeOster 

> @petercad - please take a look at the ci errors

The Linux errors aren't actually issues with the samples -- the errors come from oneMKL itself and look like a configuration issue on the CI machine. Will look into it.

**Comment by JoeOster** [comment]
> @JoeOster
> 
> > @petercad - please take a look at the ci errors
> 
> The Linux errors aren't actually issues with the samples -- the errors come from oneMKL itself and look like a configuration issue on the CI machine. Will look into it.

@mkitez -can you look into this please

**Comment by petercad** [comment]
> This is obviously not following the direct coding Samples guidelines, but does seem to follow a consistent oneMKL coding style.
> 
> Looks good for now, I'd expect more DPC++ style changes recommened after the next beta update.

Thanks, @pmpeter1 for the review! We'll continue working on the DPC++ styling. I also have some style updates from our team for the block_cholesky and block_lu samples that I will push shortly.

**Comment by JoeOster** [comment]
@petercad - Please take a look at the ci errors. They all revolve around one issue for Linux samples
Intel MKL FATAL ERROR: Error on loading function 'zeDeviceGetProperties'.

**Comment by petercad** [comment]
> @petercad - Please take a look at the ci errors. They all revolve around one issue for Linux samples
> Intel MKL FATAL ERROR: Error on loading function 'zeDeviceGetProperties'.

Already done. It's not an issue in the samples, it's a configuration on the CI machine (missing libze_loader.so symlink). Waiting for machine to be fixed.

**Pull request merged**